iml-api/src/graphql.rs

```diff
@@ -143,6 +144,94 @@ impl QueryRoot {
 
         Ok(xs)
     }
+    #[graphql(arguments(
+        limit(description = "paging limit, defaults to 20"),
+        offset(description = "Offset into items, defaults to 0"),
+        dir(description = "Sort direction, defaults to asc"),
+        args(description = "Snapshot listing arguments")
+    ))]
+    /// Fetch the list of snapshots
+    async fn snapshots(
+        context: &Context,
+        limit: Option<i32>,
+        offset: Option<i32>,
+        dir: Option<SortDir>,
+        args: List,
+    ) -> juniper::FieldResult<Vec<Snapshot>> {
+        let dir = dir.unwrap_or_default();
+
+        if args.detail {
```

Comment:
I think there is no need to support it yet (because there is not support in DB yet, and actual listing is extremely slow). And in any case, `.details[]` exists: by default it includes only one record with `snapshot_role: None` (I believe it is MDT). In a detailed listing it would include components of the snapshot with `snapshot_role` like `OST0001`.

Reply:
How is there no support in DB if the `details` field of `Snapshot` is saved to DB? So we either ask for them or not.

Reply:
I think `detail` is being overloaded. 

A non-detail call has this result:

```
lctl snapshot_list -F zfsmo --name zfsmo-snapshot1

filesystem_name: zfsmo
snapshot_name: zfsmo-snapshot1
create_time: Wed Sep  2 15:29:16 2020
modify_time: Wed Sep  2 15:29:16 2020
snapshot_fsname: 462af955
status: not mount
```

Where a detail call looks like:

```
[root@mds1 ~]# lctl snapshot_list -F zfsmo --name zfsmo-snapshot1 --detail

filesystem_name: zfsmo
snapshot_name: zfsmo-snapshot1

snapshot_role: MGS
snapshot_fsname: 462af955
modify_time: Wed Sep  2 15:29:16 2020
create_time: Wed Sep  2 15:29:16 2020
status: not mount

snapshot_role: MDT0000
create_time: Wed Sep  2 15:29:16 2020
modify_time: Wed Sep  2 15:29:16 2020
snapshot_fsname: 462af955
status: not mount

...
```

Both calls have the same fields, the difference is the detail call is listing each target. 

Also, details will be empty if the MGT / MDT are not co-located on the same node.

Reply:
In addition, I wonder if this struct should not be nested as the result of this API call.